test/interpolating_tests.jl

```diff
@@ -55,10 +55,14 @@ function callback_saving(u, t, integrator, sol)
     temp = sol(t)
     return vjp((x) -> lotka_volterra(temp, x, t), integrator.p, u)[1]
 end
+function callback_saving_inplace(du, u, t, integrator, sol)
+    temp = sol(t)
+    du .= vjp((x) -> lotka_volterra(temp, x, t), integrator.p, u)[1]
```

Comment:
you have the allocation right here, so it's not saving an allocation because of how you've written the integrand function. At least it's up to the user though.